core/api_contacts.go

```diff
@@ -88,26 +109,42 @@ func (a *api) searchContacts(g *gin.Context) {
 	if err != nil {
 		wait = 5
 	}
-	add, err := strconv.ParseBool(opts["add"])
-	if err != nil {
-		add = false
-	}
 
-	query := &ContactInfoQuery{
+	query := &ContactQuery{
 		Id:       opts["peer"],
 		Address:  opts["address"],
 		Username: opts["username"],
 		Local:    local,
 		Limit:    limit,
 		Wait:     wait,
-		Add:      add,
```

Comment:
This was a silly way to add a contact... and greatly complicated the search handler. Removed.